nemo_curator/distributed_data_classification/domain_classifier_inference.py

```diff
@@ -17,162 +17,130 @@
-)
-from nemo_curator.utils.distributed_utils import (
-    get_client,
-    load_object_on_worker,
-    process_all_batches,
-    read_data,
-    write_to_disk,
-)
+from nemo_curator.utils.distributed_utils import get_client, read_data, write_to_disk
 from nemo_curator.utils.file_utils import get_remaining_files
 
 warnings.filterwarnings("ignore")
 
 
-def inference_per_partition(
-    df,
-    max_chars,
-    batch_size,
-    num_workers,
-    model_file_name,
-    labels,
-    autocast,
-):
-    """
-    This function runs domain classification on a subset of the data.
-    It loads the CFG, a data iterator, and then calls the `process_all_batches` function,
-    which loads the domain classifier and runs inference.
-
-    Args:
-        df: A Dask DataFrame partition with a "text" column and a dummy "pred" column.
-        max_chars: The maximum number of characters allowed in the truncated text.
-        batch_size: How many samples per batch to load with PyTorch DataLoader.
-        num_workers: How many subprocesses to use for PyTorch DataLoader.
-        model_file_name: The path to the model file.
-        labels: The list of domain labels.
-        autocast: A boolean representing whether to perform inference with mixed precision.
-    Returns:
-        The input Dask DataFrame with the calculated "pred" column.
-
-    """
-    cfg = cfg_per_partition()
-
-    dataset_valid = TestDataset(cfg, df, max_chars)
-    loader_valid = torch.utils.data.DataLoader(
-        dataset_valid,
-        batch_size=batch_size,
-        shuffle=False,
-        num_workers=num_workers,
-    )
-    device = torch.device("cuda")
-    load_model_kwargs = {"cfg": cfg, "device": device, "model_path": model_file_name}
-    run_inference_kwargs = {"autocast": autocast}
-    st = time.time()
-    preds = process_all_batches(
-        loader_valid,
-        load_model,
-        load_model_kwargs,
-        run_inference,
-        run_inference_kwargs,
-    )
-    preds = preds.cpu().numpy()
-    df["pred"] = [labels[i] for i in preds]
-
-    et = time.time()
-    print(
-        f"Time taken for inference for num_batches: {len(loader_valid)} : {et-st} s",
-        flush=True,
-    )
-
-    return df
+@dataclass
+class Config:
+    model = "microsoft/deberta-v3-base"
+    fc_dropout = 0.2
+    max_len = 512
 
 
-def cfg_per_partition():
-    """
-    This function loads the CFG on the worker currently running the task.
-    See `load_object_on_worker` function.
-
-    Returns:
-        A CFG with a set `tokenizer` attribute.
-
-    """
-    return load_object_on_worker("cfg_with_tokenizer", load_cfg_with_tokenizer, {})
-
-
-def load_cfg_with_tokenizer():
-    """
-    This function loads the CFG needed for domain classification.
-
-    Returns:
-        A CFG with a set `tokenizer` attribute.
-
-    """
-    cfg = CFG()
-    tokenizer = DebertaV2TokenizerFast.from_pretrained(cfg.model)
-    cfg.tokenizer = tokenizer
-    return cfg
+class CustomModel(nn.Module):
```

Comment:
I think we should just remove this file in favor of API example or edit it to use API directly.  Dont think we should have 2  implementations .

Reply:
Yeah, I'm ok with that. The original intention with the `*_classifier_inference.py` scripts was to give the user some insight on how to write a custom MNMG script for their own LLM classifiers (that they had trained themselves). But I think we're moving away from that with the API anyway, and the user can always dig into our other MNMG code if they're really interested.

Reply:
Please take another look now.